src/cmr/sizing/core.clj

```diff
@@ -22,32 +22,81 @@
 ;;; Note that all internal operations for estimating size are performed
 ;;; assuming bytes as the units.
 
-(defn get-measurement
+(defn- read-number
+  "If value is string, read-string.  Otherwise ignore."
+  [value]
+  (if (string? value)
+    (read-string value)
+    value))
+
+(defn- get-measurement
   [variable]
   (let [dims (get-in variable [:umm :Dimensions])
         total-dimensionality (reduce * (map :Size dims))
         data-type (get-in variable [:umm :DataType])]
     (* total-dimensionality
        (util/data-type->bytes data-type))))
 
+(defn- get-avg-gran-size
+  "Gets SizeEstimation value for AverageSizeOfGranulesSampled and parses it to a number."
+  [variable]
+  (-> variable
+      (get-in [:umm :SizeEstimation :AverageSizeOfGranulesSampled])
+      read-number))
+
+(defn- get-avg-compression-rate-ascii
+  "Gets SizeEstimation value for AvgCompressionRateASCII and parses it to a number."
+  [variable]
+  (-> variable
+      (get-in [:umm :SizeEstimation :AvgCompressionRateASCII])
+      read-number))
+
+(defn- get-avg-compression-rate-netcdf4
+  "Gets SizeEstimation value for AvgCompressionRateNetCDF4 and parses it to a number."
+  [variable]
+  (-> variable
+      (get-in [:umm :SizeEstimation :AvgCompressionRateNetCDF4])
+      read-number))
+
 (defn estimate-dods-size
+  "Calculates the estimated size for DODS format."
   [granule-count variables]
   (let [compression 1
         metadata 0 ; included here for symmetry
         measurements (reduce + (map get-measurement variables))]
     (+ (* granule-count compression measurements) metadata)))
 
-(defn estimate-netcdf3-size
+(defn- estimate-netcdf3-size
+  "Calculates the estimated size for NETCDF3 format."
   [granule-count variables metadata]
   (let [compression 1
         measurements (reduce + (map get-measurement variables))]
     (+ (* granule-count compression measurements) metadata)))
 
+(defn- estimate-netcdf4-or-ascii-size
+  "Calculates the estimated size for ASCII or NETCDF4 format.
+   total-granule-input-bytes is a value given by the client in the size estimate request."
+  [granule-count variables params fmt]
+  (reduce (fn [total-estimate variable]
```

Comment:
Since this function is a little long it _might_ be worth changing from an anonymous function to a defined function. I don't feel strongly about it either way.

Reply:
I feel a little more strongly about that sort of thing than Chris ;-) We should ...